Sources/_StringProcessing/Engine/Processor.swift

```diff
@@ -339,6 +340,107 @@ extension Processor {
     }
   }
 
+  mutating func regexAssert(
+    by kind: AST.Atom.AssertionKind,
+    _ anchorsMatchNewlines: Bool,
+    _ usesSimpleUnicodeBoundaries: Bool,
+    _ usesASCIIWord: Bool,
+    _ semanticLevel: MatchingOptions.SemanticLevel
+  ) throws -> Bool {
+    // Future work: Optimize layout and dispatch
+    
+    // FIXME: Depends on API model we have... We may want to
+    // think through some of these with API interactions in mind
+    //
+    // This might break how we use `bounds` for both slicing
+    // and things like `firstIndex`, that is `firstIndex` may
+    // need to supply both a slice bounds and a per-search bounds.
```

Comment:
I think this stuff got fixed when @natecook1000 added both subject bounds and search bounds.